src/WebJobs.Script.WebHost/Configuration/AppServiceOptionsChangeTokenSource.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    internal class AppServiceOptionsChangeTokenSource : IOptionsChangeTokenSource<AppServiceOptions>
```

Comment:
Can you go ahead and make this generic? I'd like to use it in v3 :-)

`SpecializationChangeTokenSource<TOptions> : IOptionsChangeTokenSource<TOptions>`... will that work?

Reply:
... and add a comment that "An implementation of `IOptionsChangeTokenSource<TOptions>` that automatically signals its change token when specialization occurs."